crypto/asn1/a_print.c

```diff
@@ -25,24 +25,16 @@ int ASN1_PRINTABLE_type(const unsigned char *s, int len)
 
     while ((*s) && (len-- != 0)) {
         c = *(s++);
-#ifndef CHARSET_EBCDIC
-        if (!(((c >= 'a') && (c <= 'z')) ||
-              ((c >= 'A') && (c <= 'Z')) ||
-              ((c >= '0') && (c <= '9')) ||
-              (c == ' ') || (c == '\'') ||
-              (c == '(') || (c == ')') ||
-              (c == '+') || (c == ',') ||
-              (c == '-') || (c == '.') ||
-              (c == '/') || (c == ':') || (c == '=') || (c == '?')))
+        if (!(ossl_isalnum(c)
+              || c == ' ' || c == '\''
+              || c == '(' || c == ')'
+              || c == '+' || c == ','
+              || c == '-' || c == '.'
+              || c == '/' || c == ':'
+              || c == '=' || c == '?'))
```

Comment:
This check is the same check as in a_mbstr.c:352, perhaps this can be tagged as another type (e.g. CTYPE_ASN1_PRINTABLE)?